lib/DiscoveryStreamFeed.jsm

```diff
@@ -537,13 +560,30 @@ this.DiscoveryStreamFeed = class DiscoveryStreamFeed {
     this.writeImpressionsPref(PREF_SPOC_IMPRESSIONS, impressions);
   }
 
+  recordTopRecImpressions(recId) {
+    let impressions = this.readImpressionsPref(PREF_REC_IMPRESSIONS);
+    impressions = {...impressions, [recId]: Date.now()};
```

Comment:
If we want to expire based on first seen time, we should check
```js
if (!impressions[recId]) {
  impressions[recId] = Date.now();
  writeImpressions()
}
```
Not sure if we really need to spread for a new object in this situation, but I see that's how spocs work.

Reply:
I think you're right but something's up with github making it hard to confirm, give me a sec.

Reply:
you're right, this is a bug, it should filter out items already in the impressions and not overwrite them.